File: js/map_loader.ts
import fs from "fs"

export type ObjectPlayer = {
    name: "Player"
    position: [number, number]
}

export type ObjectCloud = {
    name: "Cloud"
    position: [number, number]
    size: [number, number]
    hash: string
    color: number
}

export type ObjectWind = {
    name: "Wind"
    position: [number, number]
}

export type ObjectTrampoline = {
    name: "Trampoline"
    position: [number, number]
}

export type ObjectHelicopter = {
    name: "Helicopter"
    position: [number, number]
}

export type Object = ObjectPlayer | ObjectCloud | ObjectWind | ObjectTrampoline | ObjectHelicopter

export type Map = {
    size: [number, number]
    objects: Object[]
}

export function loadMap(): Map {
    const content = fs.readFileSync("assets/map/Map.json").toString()
    const json = JSON.parse(content) as Map
    
    return json
}

File: go/map_loader.go
package main

import (
	"encoding/json"
	"os"
)

type Object struct {
	Name string			`json:"name"`
    Position []float32  `json:"position"`
    Hash string         `json:"hash"`
}

type Map struct {
	Size []float32		`json:"size"`
	Objects []Object	`json:"objects"`
}

func loadMap() Map {
	data, err := os.ReadFile("../assets/map/Map.json")
	
	if err != nil {
		panic("Gak nemu map file")
	}
	
	theMap := Map{}
	
	err = json.Unmarshal(data, &theMap)
	
	if err != nil {
		panic("Format map gak bener")
	}
	
	return theMap
}